dataland-internal-storage/src/main/kotlin/org/dataland/datalandinternalstorage/services/DatabaseDataStore.kt

```diff
@@ -83,12 +85,22 @@ class DatabaseDataStore(
         }
     }
 
+    /**
+     * Stores a Data Item while ensuring that there is no active transaction. This will guarantee that the write
+     * is commited after exit of this method.
+     * @param dataItem the DataItem to be stored
+     */
+    @Transactional(propagation = Propagation.NEVER)
+    fun storeDataItemWithoutTransaction(dataItem: DataItem) {
+        dataItemRepository.save(dataItem)
+    }
+
     /**
      * Reads data from a database
      * @param dataId the id of the data to be retrieved
      * @return the data as json string with id dataId
      */
     fun selectDataSet(dataId: String): String {
-        return dataItemRepository.findById(dataId).orElse(DataItem("", "")).data
+        return dataItemRepository.findById(dataId).orElse(DataItem("", "\"\"")).data
```

Comment:
Was that even right behaviour in the first place? Shouldn't we return a 404 in the case a data Id is not present - not not simply return an empty object?

Reply:
I would also prefer that.

Reply:
Done

Reply:
As agreed: 
* don't use 404, but HttpStatus.NOT_FOUND
* Throw a ResourceNotFoundException if the optional from database is not present

Reply:
ok